core/src/net/mod.rs

```diff
@@ -1137,6 +1171,8 @@ pub mod net_test_helpers {
                             let ping = BigPingMsg::new(pong.i + 1, self.data_size);
                             self.target.tell(ping, self);
                         }
+                    } else if let Some(promise) = self.promise.take() {
+                        promise.fulfil(()).expect("Failed to fulfil promise");
```

Comment:
Seeing all these `fulfil(())` all over the codebase, I'm starting to think I should add a utility function to `Promise<()>` so you can just call say `promise.complete()`. Would make stuff a bit more readable, perhaps. I'll put on my list^^